web/router.go

```diff
@@ -38,10 +43,64 @@ func SetupRoutes(router *gin.Engine) {
 	router.Use(middlewares.ParseHost())
 	router.Use(middlewares.ServeApp(apps.Serve))
 	router.Use(middlewares.ErrorHandler())
+	router.Use(corsMiddleware("/apps", "/data", "/files"))
```

Comment:
Why did you insert the middleware here, and not like `middlewares.NeedInstance` in each `Group`?